aslam_cv_pipeline/src/visual-npipeline.cc

```diff
@@ -271,7 +290,8 @@ void VisualNPipeline::work(size_t camera_index, const cv::Mat& image, int64_t ti
           num_consecutive_complete = 0u;
         }
         if (num_consecutive_complete >= kNumMinConsecutiveCompleteThreshold) {
-          delete_upto_including_index = idx - kNumMinConsecutiveCompleteThreshold;
+          delete_upto_including_index = static_cast<int>(
+              idx - kNumMinConsecutiveCompleteThreshold);
```

Comment:
hm this doesn't work I believe...
if idx == 1u and kNumMinConsecutiveCompleteThreshold == 2u (which can happen),
then delete_upto_including_index will be something like 1237239847398478347.
cast to int has to happen individually before the subtraction.

Reply:
This modification does not change anything because it was an implicit cast before (so they same thing would have happened). However, you are right that this is potentially dangerous. From the C++11 standard 4.7 clause 3: *If the destination type is signed, the value is unchanged if it can be represented in the destination type (and bit-field width); otherwise, the value is **implementation-defined**.* It will probably always work on all of our machines but *implementation-defined* is not the same as *defined*.